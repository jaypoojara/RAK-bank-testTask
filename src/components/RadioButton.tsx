import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import {TouchableOpacityProps} from 'react-native-gesture-handler';
import {colors} from '../utils/theme';

interface IProps extends TouchableOpacityProps {
  value?: boolean;
  onChange?: (arg: boolean) => void;
}

const RadioButton = (props: IProps) => {
  const {value = false, onChange} = props;
  return (
    <TouchableOpacity {...props} onPress={() => onChange?.(!value)}>
      <View style={styles.container}>
        {value && <View style={styles.fill} />}
      </View>
    </TouchableOpacity>
  );
};

export default RadioButton;

const styles = StyleSheet.create({
  container: {
    height: 30,
    width: 30,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: colors.primary,
    justifyContent: 'center',
    alignItems: 'center',
  },
  fill: {
    height: '80%',
    width: '80%',
    borderRadius: 5,
    backgroundColor: colors.primary,
  },
});
