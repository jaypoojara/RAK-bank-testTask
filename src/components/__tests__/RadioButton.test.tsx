import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import RadioButton from '../RadioButton';

describe('RadioButton', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<RadioButton />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows filled circle when value is true', () => {
    const {getByTestId} = render(<RadioButton value={true} />);
    const fill = getByTestId('radio-fill');
    expect(fill).toBeTruthy();
  });

  it('shows empty circle when value is false', () => {
    const {queryByTestId} = render(<RadioButton value={false} />);
    const fill = queryByTestId('radio-fill');
    expect(fill).toBeNull();
  });

  it('calls onChange when pressed', () => {
    const onChangeMock = jest.fn();
    const {getByTestId} = render(
      <RadioButton value={false} onChange={onChangeMock} />,
    );
    fireEvent.press(getByTestId('radio-button'));
    expect(onChangeMock).toHaveBeenCalledWith(true);
  });

  it('does not call onChange if no onChange prop is provided', () => {
    const {getByTestId} = render(<RadioButton value={false} />);
    fireEvent.press(getByTestId('radio-button'));
  });
});
