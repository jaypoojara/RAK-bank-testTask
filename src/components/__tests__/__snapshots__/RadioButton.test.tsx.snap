// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="radio-button"
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#694F8E",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      }
    }
  />
</View>
`;
